content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -432,24 +447,27 @@ Since this flow is performing a redemption, you must have both an initiating flo
 Start writing your initiating flow following the same process used when writing the <a href="#write-the-createandissuemarsvoucher-flow">`CreateAndIssueMarsVoucher`</a> flow.
 
 1. Add these annotations: `@InitiatingFlow`, `@StartableByRPC`.
+
 2. Define the `RedeemBoardingTicketWithVoucherInitiator` class with a `@JsonConstructor`, `RpcStartFlowRequestParameters`, and returning a `SignedTransactionDigest`.
 3. Inject these services:
     * `FlowEngine`
     * `FlowIdentity`
     * `FlowMessaging`
     * `TransactionBuilderFactory`
-    * `IdentityService`
+    * `IdentityService`: Provides methods to retrieve `Party` and `AnonymousParty` instances.
     * `NotaryLookupService`
-    * `JsonMarshallingService`
+    * `JsonMarshallingService`: Parses arbitrary content in and out of JSON using standard, approved mappers.
     * `PersistenceService`
+
 4. Add the `@Suspendable` annotation.
 5. Encapsulate the flow implementation into a call method that returns the `SignedTransactionDigest`.
+
 6. Parse these parameters and add exceptions for when these parameters are incorrect or not present:
     * `voucherID`
     * `holder`
     * `recipientParty`
 
-7. Insert a method for finding the notary.
+7. Insert the following method for finding the notary: `notaryLookup.notaryIdentities.<notary name>()`
```

Comment:
We Should not be using `.first ` in the method for getting the notary so I changed `notaryLookup.notaryIdentities.first()` to `notaryLookup.notaryIdentities.<notary name>()` here and in the code. Is that correct or what is the name of the notary I'm looking for?

Reply:
I actually need ask Engineering about it...

Reply:
same^

Reply:
updated the tutorial with notaryLookup.getNotary(CordaX500Name.parse("O=notary, L=London, C=GB"))!!